mrmustard/lab_dev/circuit_components.py

```diff
@@ -216,8 +236,8 @@ def __rshift__(self, other: CircuitComponent) -> CircuitComponent:
         """
         msg = f"``__rshift__`` not supported between {self} and {other}, use ``__matmul__``."
 
-        wires_out = self.wires.output
-        wires_in = other.wires.input
+        wires_out = self.wires
```

Comment:
@ziofil I found the reason why we don't want `input` and `output` here (see [this discussion](https://github.com/XanaduAI/MrMustard/pull/353#discussion_r1506398824)).

Without them, right-shifting states (e.g., `Vacuum([0]) >> Vacuum([1])`) errors (says to use ``@``). 
Granted, this may be sub-optimal, but it may become annoying for people using the simulator (imagine launching a simulation on AWS with a path that says to do this contraction.. I would rather see the simulation's result in a slightly longer time than getting an error)

For this reason, considering that ``>>`` is meant to be high-level and user-friendly, I think that we should get rid of input and output here

Reply:
ok. then we can call them `wires_self` and `wires_other`.